src/web/src/main/java/tupperdate/web/routing/chats/Chats.messages.kt

```diff
@@ -1,14 +1,40 @@
 package tupperdate.web.routing.chats
 
 import com.google.cloud.firestore.Firestore
-import com.google.firebase.auth.FirebaseAuth
+import com.google.cloud.firestore.Query
 import io.ktor.application.*
 import io.ktor.http.*
 import io.ktor.response.*
 import io.ktor.routing.*
+import tupperdate.common.dto.ConversationDTO
+import tupperdate.common.dto.MessageDTO
 import tupperdate.web.auth.firebaseAuthPrincipal
 import tupperdate.web.exceptions.statusException
+import tupperdate.web.model.*
+import tupperdate.web.util.await
 
 fun Route.getMessages(store: Firestore) = get("/{userId}/messages") {
-    call.respond(HttpStatusCode.NotImplemented)
+    val uid = call.firebaseAuthPrincipal?.uid ?: statusException(HttpStatusCode.Unauthorized)
+    val userId = call.parameters["userId"] ?: statusException(HttpStatusCode.BadRequest)
+
+    val chatId = minOf(uid, userId) + "_" + maxOf(uid, userId)
+    val chat = store.collection("chats").document(chatId).get().await().toObject(Chat::class.java) ?: statusException(HttpStatusCode.NotFound)
+
+    if (chat.user1Recipes == null || chat.user2Recipes == null) {
+        statusException(HttpStatusCode.NotFound)
+    }
+
+    val messages = store.collection("chats/${chatId}/messages")
```

Comment:
Not `store.collection(..).document(..).collection(..)` ?

Reply:
You're right, that's what i had found online and didn't question it